### PR TITLE
Add failing tests for node fragments and renamed types

### PR DESCRIPTION
**Pull request by Ambroos**
Failing tests for https://github.com/ent/ent/issues/3531

One of the added tests is in entgql/internal/todo/todo_test.go, the other is in entgql/internal/renamedtype/renamedtype_test.go. I added a new directory for renamedtype since adding a renamed type to the todo directory broke a lot of tests by changing the ID sequence.

### CODE REVIEW

File: entgql/internal/renamedtype/ent/gql_node.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	}
+	return c.noder(ctx, table, id)
+}
+
+func (c *Client) noder(ctx context.Context, table string, id int) (Noder, error) {
+	switch table {
+	case clashingtext.Table:
+		query := c.ClashingText.Query().
+			Where(clashingtext.ID(id))
+		query, err := query.CollectFields(ctx, "ClashingText")
```
Comment (Ambroos) [inline entgql/internal/renamedtype/ent/gql_node.go:105]: Manually renaming this actually fixes the test, so I think this part should be relatively easy to fix. (And I assume a similar fix is needed in gql_collection.go).
```suggestion
		query, err := query.CollectFields(ctx, "NotClashingText")
```

File: entgql/internal/renamedtype/ent/schema/clashingtext.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// Fields returns private fields.
+func (ClashingText) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("content"),
+	}
+}
+
+func (ClashingText) Annotations() []schema.Annotation {
+	return []schema.Annotation{
+		entgql.Type("NotClashingText"),
```
[the remark this answers is not in the document]
Comment (Ambroos) [reply entgql/internal/renamedtype/ent/schema/clashingtext.go:39]: I'll give it another attempt. That is what I tried originally, but it breaks a lot of tests due to all IDs shifting (global unique IDs).

[the remark this answers is not in the document]
Comment (a8m) [reply entgql/internal/renamedtype/ent/schema/clashingtext.go]: https://github.com/ent/contrib/pull/505 should handle renamed types. I'll give a look at the Node fragments later.

[the remark this answers is not in the document]
Comment (a8m) [reply entgql/internal/renamedtype/ent/schema/clashingtext.go]: https://github.com/ent/contrib/pull/507 should address the second issue.

**Comment by a8m** [comment]
Thanks for reporting this, @Ambroos ❤️. The issue was resolved by #505 and #507

**Comment by Ambroos** [comment]
I'd give you a #thanks if we both still worked for our previous employers! This is great (as is the rest of Ent), thanks for fixing this!

**Comment by a8m** [comment]
> I'd give you a #thanks if we both still worked for our previous employers! This is great (as is the rest of Ent), thanks for fixing this!

Thanks again for reporting this ❤️

**Pull request closed without merging**